setup.sh

```diff
@@ -139,9 +139,11 @@ if [[ -z $QUIET_INTRO ]]; then
         if [[ $OLD_COMMIT != $NEW_COMMIT ]]; then
             # After syncing to remote, delegate to the new setup script
             # ... in case that changed.
-            git --git-dir="$KIT_PATH/.git" --work-tree="$KIT_PATH" checkout --quiet -B adsk-setup && \
+            # [jokram] Why is the checkout of a new branch adsk-setup necessary? The reset --hard will overwrite it, or?
+            # Or is it just to mark that this was the previous version?
+            git --git-dir="$KIT_PATH/.git" --work-tree="$KIT_PATH" checkout --quiet -B $KIT_ID-setup && \
             git --git-dir="$KIT_PATH/.git" --work-tree="$KIT_PATH" reset --quiet --hard origin/$BRANCH && \
```

Comment:
I think now I got it:
`git reset --hard origin/production` will modify the current branch pointer (which might be `branch-X`) to also point to `origin/production`. And by this the `branch-X` would have been modified. So `git checkout -B adsk-setup` is just to ensure that you have temp. branch which can be modified by the reset command.